frontend/src/metabase/query_builder/components/view/NativeQueryButton.styled.jsx

```diff
@@ -21,3 +26,39 @@ export const SqlIconButton = styled(Button)`
 SqlIconButton.defaultProps = {
   icon: "sql",
 };
+
+export const NativeCodeWrapper = styled.pre`
+  box-sizing: border-box;
+  display: inline-block;
+  margin: 0;
+  padding: 0;
+  max-width: 100%;
+  overflow: hidden;
+  vertical-align: bottom;
+  width: 100%;
+`;
+
+export const NativeCodeContainer = styled.code`
+  box-sizing: border-box;
+  display: inline-block;
+  margin: 0;
+  max-height: 20vh;
+  max-width: 100%;
+  overflow: auto;
+  vertical-align: bottom;
+  white-space: pre;
+  width: 100%;
+  word-break: break-all;
+
+  ${breakpointMinHeightExtraSmall} {
```

Comment:
Height breakpoints added so we use the maximum screen `real estate` to display the native code.

Reply:
Great, I focused too much on getting this to work with only flexbox and totally forgot the good ol' media query 😆